src/python/gudhi/weighted_rips_complex.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Raphaël Tinarrage, Yuichi Ike, Masatoshi Takenouchi
+#
+# Copyright (C) 2020 Inria, Copyright (C) 2020 FUjitsu Laboratories Ltd.
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from gudhi import SimplexTree
+
+class WeightedRipsComplex:
+    """
+    class to generate a weighted Rips complex 
+    from a distance matrix and weights on vertices
+    """
+    def __init__(self, 
+                distance_matrix, 
+                weights=None,
+                max_filtration=float('inf')):
+        """
+        Parameters:
+            distance_matrix: list of list of float,
+                distance matrix (full square or lower triangular)
+            filtration_values: list of float,
+                weight for each vertex
+            max_filtration: float,
+                specifies the maximal filtration value to be considered        
+        """
+        self.distance_matrix = distance_matrix
+        if weights is not None:
+            self.weights = weights
```

Comment:
I think we could allow something like `weights="diagonal"` to mean that the weight of point i is `distance_matrix[i][i]`.